### PR TITLE
Add COinS metadata parser

### PR DESCRIPTION
**Pull request by mvolz**
Add parseCOinS() which selects all span tags with
class Z3988[1], and parses the title tag into an
object. Hierarchical keys (e.g. rft.date) are
split, with the top level key 'rft' pointing to
an object with the second level key as a key
in that object. Then the object is added to the
list, as multiple COinS objects can exist in
a page.

There is also a separate exportable parsing function
parseCOinSTitle() for the contents of the title tag, 
which can be used separately, e.g. for parsing the 
contents of a cross ref request [2].

[1] http://ocoins.info/#id3205609413
[2] http://search.crossref.org/dois?q=10.5555%2F12345678

Phabricator issue: https://phabricator.wikimedia.org/T104174

Bug: T104174

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -45,9 +45,81 @@ exports.parseAll = function(chtml){
 };
 
 /**
+ * Scrapes COinS data given Cheerio loaded html object
+ * @param  {Object}   chtml html Cheerio object
+ * @return {Object}         BBPromise for COinS metadata
+ */
+exports.parseCOinS = BBPromise.method(function(chtml){
```
Comment (d00rman) [inline lib/index.js]: Since you return a Promise, no need to wrap it with `BBPromise.method()`

File: lib/index.js [hunk 2]
```diff
@@ -45,9 +45,81 @@ exports.parseAll = function(chtml){
...
+ * @param  {String}   title String corresponding to value of title tag in span element
+ * @return {Object}         BBPromise for CoinS metadata
+ */
+exports.parseCOinSTitle = BBPromise.method(function(title){
+	var metadata = {};
+	var rft = {};
+	var value;
+	var key;
+	if (typeof title !== 'string'){
+		BBPromise.reject('Provided value must be a string; Got ' + typeof title);
```
Comment (d00rman) [inline lib/index.js]: Need to return the rejection for it to have an effect

File: lib/index.js [hunk 3]
```diff
@@ -45,9 +45,81 @@ exports.parseAll = function(chtml){
...
+		if (key.length === 2){ // Split key e.g. rft.date
+			if (key[0] !== 'rft') {return;} // Invalid hierarchical key
+			rft[key[1]] = value; // Add rft value to rft key - this will overwrite duplicates, if they exist
+		}
+	});
+	if (Object.keys(rft).length){ // Add rft object if it is not empty
+		metadata.rft = rft;
+	}
+	if (!Object.keys(metadata).length){
+		BBPromise.reject('No COinS in provided string');
```
Comment (d00rman) [inline lib/index.js]: Idem - `return`

File: lib/index.js (hunk 1 above)
Comment (mvolz) [inline lib/index.js]: Done

File: lib/index.js (hunk 2 above)
Comment (mvolz) [inline lib/index.js]: Done

File: lib/index.js (hunk 3 above)
Comment (mvolz) [inline lib/index.js]: Done

**Pull request merged**